src/transpose.cpp

```diff
@@ -0,0 +1,140 @@
+#include <Rcpp.h>
+using namespace Rcpp;
+
+// copy from x with number of rows, to c column in columns
+template <typename T, class RTYPE>
```

Comment:
Felt the same. `copy_column` or `pluck_column`? I'm going with `copy_column` for now.